test/unit/BPool/BPool.t.sol

```diff
@@ -50,4 +72,203 @@ contract BPool is BPoolBase {
+    // it should revert
+    vm.expectRevert(IBPool.BPool_Reentrancy.selector);
+    bPool.getCurrentTokens();
+  }
+
+  function test_GetCurrentTokensWhenPreconditionsAreMet() external view {
+    // it returns pool tokens
+    address[] memory _tokens = bPool.getCurrentTokens();
+    assertEq(_tokens.length, tokens.length);
+    assertEq(_tokens[0], tokens[0]);
+    assertEq(_tokens[1], tokens[1]);
+  }
+
+  function test_GetDenormalizedWeightRevertWhen_ReentrancyLockIsSet() external {
+    bPool.call__setLock(_MUTEX_TAKEN);
+    // it should revert
+    vm.expectRevert(IBPool.BPool_Reentrancy.selector);
+    bPool.getDenormalizedWeight(tokens[0]);
+  }
+
+  function test_GetDenormalizedWeightRevertWhen_TokenIsNotBound() external {
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bPool.getDenormalizedWeight(unknownToken);
+  }
+
+  function test_GetDenormalizedWeightWhenPreconditionsAreMet() external view {
+    // it returns token weight
+    uint256 _tokenWeight = bPool.getDenormalizedWeight(tokens[0]);
+    assertEq(_tokenWeight, tokenWeight);
+  }
+
+  function test_GetTotalDenormalizedWeightRevertWhen_ReentrancyLockIsSet() external {
+    bPool.call__setLock(_MUTEX_TAKEN);
+    // it should revert
+    vm.expectRevert(IBPool.BPool_Reentrancy.selector);
+    bPool.getTotalDenormalizedWeight();
+  }
+
+  function test_GetTotalDenormalizedWeightWhenPreconditionsAreMet() external view {
+    // it returns total weight
+    uint256 _totalWeight = bPool.getTotalDenormalizedWeight();
+    assertEq(_totalWeight, totalWeight);
+  }
+
+  function test_GetNormalizedWeightRevertWhen_ReentrancyLockIsSet() external {
+    bPool.call__setLock(_MUTEX_TAKEN);
+    // it should revert
+    vm.expectRevert(IBPool.BPool_Reentrancy.selector);
+    bPool.getNormalizedWeight(tokens[0]);
+  }
+
+  function test_GetNormalizedWeightRevertWhen_TokenIsNotBound() external {
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bPool.getNormalizedWeight(unknownToken);
+  }
+
+  function test_GetNormalizedWeightWhenPreconditionsAreMet() external view {
+    // it returns normalized weight
+    //     normalizedWeight = tokenWeight / totalWeight
+    uint256 _normalizedWeight = bPool.getNormalizedWeight(tokens[0]);
+    assertEq(_normalizedWeight, 0.1e18);
+  }
+
+  function test_GetBalanceRevertWhen_ReentrancyLockIsSet() external {
+    bPool.call__setLock(_MUTEX_TAKEN);
+    // it should revert
+    vm.expectRevert(IBPool.BPool_Reentrancy.selector);
+    bPool.getBalance(tokens[0]);
+  }
+
+  function test_GetBalanceRevertWhen_TokenIsNotBound() external {
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bPool.getBalance(unknownToken);
+  }
+
+  function test_GetBalanceWhenPreconditionsAreMet(uint256 tokenBalance) external {
+    vm.mockCall(tokens[0], abi.encodePacked(IERC20.balanceOf.selector), abi.encode(tokenBalance));
```

Comment:
consider doing this in setup, it's repeated

Reply:
hmm not possible because of the setters